csrc/megatron/fused_weight_gradient_dense_cuda.cu

```diff
@@ -10,10 +10,285 @@
 /* Includes, cuda */
 #include <cublas_v2.h>
 #include <cuda_runtime.h>
-
 #include "type_shim.h"
 
+/* Includes, HIP */
+#include <hipblaslt/hipblaslt-ext.hpp>
+
+#ifndef CHECK_HIP_ERROR
+#define CHECK_HIP_ERROR(error)                    \
+    if(error != hipSuccess)                       \
+    {                                             \
+        fprintf(stderr,                           \
+                "Hip error: '%s'(%d) at %s:%d\n", \
+                hipGetErrorString(error),         \
+                error,                            \
+                __FILE__,                         \
+                __LINE__);                        \
+        exit(EXIT_FAILURE);                       \
+    }
+#endif
+
+#ifndef CHECK_HIPBLASLT_ERROR
+#define CHECK_HIPBLASLT_ERROR(error)                                                      \
+    if(error != HIPBLAS_STATUS_SUCCESS)                                                   \
+    {                                                                                     \
+        fprintf(stderr, "hipBLASLt error(Err=%d) at %s:%d\n", error, __FILE__, __LINE__); \
+        fprintf(stderr, "\n");                                                            \
+        exit(EXIT_FAILURE);                                                               \
+    }
+#endif
+
+// BF16 Tensor core wrapper around cublas GEMMEx
+void gemmex_wrapper(
+    hipblasLtHandle_t handle,
+    hipblasOperation_t transa,
+    hipblasOperation_t transb,
+    int m,
+    int n,
+    int k,
+    int batch_count,
+    float& alpha,
+    float& beta,
+    at::BFloat16* A,
+    at::BFloat16* B,
+    float*    C,
+    float*    D,
+    void*     d_workspace,
+    int64_t   max_workspace_size,
+    hipStream_t   stream) {
+
+    hipblaslt_ext::GemmPreference gemmPref;
+    gemmPref.setMaxWorkspaceBytes(max_workspace_size);
+    hipblaslt_ext::Gemm gemm(
+        handle, transa, transb, HIP_R_16BF, HIP_R_16BF, HIP_R_32F, HIP_R_32F, HIPBLAS_COMPUTE_32F);
+
+    hipblaslt_ext::GemmEpilogue
```

Comment:
This is using hipblasLtExt API. We haven't used that a lot. We mostly use hipblasLt API. I will need to check with hipblaslt team regarding the difference (in performance implication). And as hipblasLt API is more commonly used, I expect it would be more stable and have fewer bugs.